plugin/src/main/scala/com/geirsson/CiReleasePlugin.scala

```diff
@@ -145,27 +147,45 @@ object CiReleasePlugin extends AutoPlugin {
         // https://github.com/olafurpg/sbt-ci-release/issues/64
         val reloadKeyFiles =
           "; set pgpSecretRing := pgpSecretRing.value; set pgpPublicRing := pgpPublicRing.value"
-        if (!isTag) {
-          if (isSnapshotVersion(currentState)) {
-            println(s"No tag push, publishing SNAPSHOT")
+
+        if (shouldDeployToSonatypeCentral) {
+          if (isSnapshot) {
+            println(s"Sonatype Central does not accept snapshots, only official releases. Aborting release.")
+            currentState
+          } else if (!isTag) {
+            println(s"No tag published. Cannot publish an official release without a tag and Sonatype Central does not accept snapshot releases. Aborting release.")
+            currentState
+          } else {
+            println("Tag push detected, publishing a stable release")
             reloadKeyFiles ::
-              sys.env.getOrElse("CI_SNAPSHOT_RELEASE", "+publish") ::
+              sys.env.getOrElse("CI_CLEAN", "; clean ; sonatypeBundleClean") ::
+              sys.env.getOrElse("CI_RELEASE", "+publishSigned") ::
+              sys.env.getOrElse("CI_SONATYPE_RELEASE", "sonatypeCentralRelease") ::
```

Comment:
Maybe this switch should be located in sbt-sonatype rather than human having to change the command depending on the context.

Reply:
I personally agree and originally implemented exactly that, but this part was rejected (see https://github.com/xerial/sbt-sonatype/pull/474#issuecomment-1970207924). @xerial preferred separating out the commands and leaving it to downstream repos to handle as they see fit. To be clear, the command will not need to change. With this change, users will simply need to add `ThisBuild / sonatypeCredentialHost := sonatypeCentralHost` to `build.sbt` and `ci-release` will handle everything else. I will add a note to the docs to clarify that.

Reply:
@eed3si9n , is this a blocker or does it require further support/discussion?

Reply:
The fact that the downstream plugin has to put this switch here indicates that this is a leaky abstraction, but it is what it is I guess. LGTM. Thanks for your patience.